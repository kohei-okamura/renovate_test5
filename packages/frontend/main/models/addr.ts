/*
 * Copyright © 2019 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { Prefecture } from '@zinger/enums/lib/prefecture'

/**
 * 住所.
 */
export type Addr = Readonly<{
  /** 郵便番号 */
  postcode: string

  /** 都道府県 */
  prefecture: Prefecture

  /** 市区町村 */
  city: string

  /** 町名・番地 */
  street: string

  /** 建物名など */
  apartment: string
}>
