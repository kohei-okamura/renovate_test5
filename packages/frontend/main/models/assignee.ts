/*
 * Copyright © 2019 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */

import { StaffId } from '~/models/staff'

/**
 * 予実担当スタッフ.
 */
export type Assignee = Readonly<{
  /** スタッフID */
  staffId: StaffId | undefined

  /** 未定フラグ */
  isUndecided: boolean

  /** 研修フラグ */
  isTraining: boolean
}>
