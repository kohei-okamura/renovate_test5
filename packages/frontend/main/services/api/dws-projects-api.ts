/*
 * Copyright © 2020 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { AxiosInstance } from 'axios'
import { DateLike } from '~/models/date'
import { DwsProject, DwsProjectId } from '~/models/dws-project'
import { DwsProjectProgram } from '~/models/dws-project-program'
import { OfficeId } from '~/models/office'
import { StaffId } from '~/models/staff'
import { api, Api } from '~/services/api/core'

/**
 * 利用者：障害福祉サービス計画 API.
 */
export namespace DwsProjectsApi {
  export type Form = {
    officeId: OfficeId
    staffId: StaffId
    writtenOn: DateLike
    effectivatedOn: DateLike
    requestFromUser: string
    requestFromFamily: string
    objective: string
    programs: Writable<DwsProjectProgram>[]
  }

  type UserId = {
    userId: number
  }

  export type CreateParams = Api.CreateParams<Form> & UserId

  export type GetParams = Api.GetParams & UserId
  export type GetResponse = {
    dwsProject: DwsProject
  }

  export type UpdateParams = Api.UpdateParams<Form> & UserId
  export type UpdateResponse = GetResponse

  export type Definition =
    Api.Create<Form, CreateParams> &
    Api.Get<GetResponse, GetParams> &
    Api.Update<Form, UpdateParams, UpdateResponse>

  const endpoint = (userId: UserId['userId'], id?: DwsProjectId) => api.endpoint('users', userId, 'dws-projects', id)

  export const create = (axios: AxiosInstance): Definition => ({
    async create ({ form, userId }) {
      await axios.post(endpoint(userId), form)
    },
    async get ({ id, userId }) {
      return await api.extract(axios.get(endpoint(userId, id)))
    },
    async update ({ form, id, userId }) {
      return await api.extract(axios.put(endpoint(userId, id), form))
    }
  })
}
