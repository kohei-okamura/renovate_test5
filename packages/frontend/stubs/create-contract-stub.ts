/*
 * Copyright Â© 2019 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { ContractStatus } from '@zinger/enums/lib/contract-status'
import { DwsServiceDivisionCode } from '@zinger/enums/lib/dws-service-division-code'
import { LtcsExpiredReason } from '@zinger/enums/lib/ltcs-expired-reason'
import { ServiceSegment } from '@zinger/enums/lib/service-segment'
import { range } from '@zinger/helpers'
import { Contract, ContractId } from '~/models/contract'
import { UserId } from '~/models/user'
import { SEEDS, STUB_DEFAULT_SEED } from '~~/stubs'
import { OFFICE_ID_MAX, OFFICE_ID_MIN } from '~~/stubs/create-office-stub'
import { USER_ID_MAX, USER_ID_MIN } from '~~/stubs/create-user-stub'
import { createFaker } from '~~/stubs/fake'
import ramenIpsum from '~~/stubs/fake/ramen-ipsum'

export const CONTRACT_ID_MAX = USER_ID_MAX * 10 + 9
export const CONTRACT_ID_MIN = USER_ID_MIN * 10

export function createContractStub (id: ContractId = CONTRACT_ID_MIN): Contract {
  const userId = Math.floor(id / 10)
  const seed = STUB_DEFAULT_SEED + `${userId}`.padStart(4, '0') + `${id}`.padStart(4, '0')
  const faker = createFaker(seed)
  const ramen = ramenIpsum.factory(seed)
  const serviceSegment = faker.randomElement([
    ServiceSegment.disabilitiesWelfare,
    ServiceSegment.longTermCare
  ])
  const status = faker.randomElement(ContractStatus.values)
  const contractedOn = status !== ContractStatus.provisional
    ? faker.randomDate()
    : undefined
  const terminatedOnDays = faker.intBetween(100, 2000)
  const terminatedOn = status === ContractStatus.terminated
    ? contractedOn?.plus({ days: terminatedOnDays })
    : undefined
  const expiredReason = !(serviceSegment !== ServiceSegment.longTermCare || status !== ContractStatus.terminated)
    ? faker.randomElement(LtcsExpiredReason.values)
    : LtcsExpiredReason.unspecified
  const createPeriod = (segment: ServiceSegment) => {
    if (serviceSegment === segment && status !== ContractStatus.provisional) {
      const start = contractedOn?.plus({ days: faker.intBetween(1, 99) })
      const end = start?.plus({ days: faker.intBetween(100, terminatedOnDays) })
      return { start: start?.toISODate(), end: end?.toISODate() }
    } else {
      return { start: undefined, end: undefined }
    }
  }
  const dwsPeriods = {
    [DwsServiceDivisionCode.homeHelpService]: createPeriod(ServiceSegment.disabilitiesWelfare),
    [DwsServiceDivisionCode.visitingCareForPwsd]: createPeriod(ServiceSegment.disabilitiesWelfare)
  }
  const ltcsPeriod = createPeriod(ServiceSegment.longTermCare)
  return {
    id,
    userId,
    officeId: faker.intBetween(OFFICE_ID_MIN, OFFICE_ID_MAX),
    serviceSegment,
    status,
    contractedOn: contractedOn?.toISODate(),
    terminatedOn: terminatedOn?.toISODate(),
    dwsPeriods,
    ltcsPeriod,
    expiredReason,
    note: ramen.ipsum(255),
    createdAt: faker.randomDateTimeString(),
    updatedAt: faker.randomDateTimeString()
  }
}

export function createContractStubsForUser (userId: UserId): Contract[] {
  const count = userId === USER_ID_MIN ? 5 : createFaker(SEEDS[userId - 1]).intBetween(0, 2)
  return range(0, count).map(i => createContractStub(userId * 10 + i))
}
