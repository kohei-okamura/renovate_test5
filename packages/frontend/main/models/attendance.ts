/*
 * Copyright © 2020 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { ServiceOption } from '@zinger/enums/lib/service-option'
import { Task } from '@zinger/enums/lib/task'
import { Assignee } from '~/models/assignee'
import { ContractId } from '~/models/contract'
import { DateLike } from '~/models/date'
import { Duration } from '~/models/duration'
import { OfficeId } from '~/models/office'
import { Schedule } from '~/models/schedule'
import { ShiftId } from '~/models/shift'
import { StaffId } from '~/models/staff'
import { UserId } from '~/models/user'

/**
 * 勤務実績 ID.
 */
export type AttendanceId = number

/**
 * 勤務実績.
 */
export type Attendance = Readonly<{
  /** 勤務実績 ID */
  id: AttendanceId

  /** 勤務シフト ID */
  shiftId: ShiftId | undefined

  /** 契約 ID */
  contractId: ContractId | undefined

  /** 事業所 ID */
  officeId: OfficeId

  /** 利用者 ID */
  userId: UserId | undefined

  /** 管理スタッフ ID */
  assignerId: StaffId

  /** 予実区分 */
  task: Task

  /** サービスコード */
  serviceCode: string

  /** 頭数 */
  headcount: number

  /** 担当スタッフ */
  assignees: Assignee[]

  /** スケジュール */
  schedule: Schedule

  /** 予実所要時間 */
  durations: Duration[]

  /** オプション */
  options: ServiceOption[]

  /** 備考 */
  note: string

  /** 確定フラグ */
  isConfirmed: boolean

  /** キャンセルフラグ */
  isCanceled: boolean

  /** キャンセル理由 */
  reason: string | undefined

  /** 登録日時 */
  createdAt: DateLike

  /** 更新日時 */
  updatedAt: DateLike
}>
