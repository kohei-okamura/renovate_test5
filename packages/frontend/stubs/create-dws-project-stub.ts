/*
 * Copyright Â© 2019 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { DayOfWeek } from '@zinger/enums/lib/day-of-week'
import { DwsProjectServiceCategory } from '@zinger/enums/lib/dws-project-service-category'
import { Recurrence } from '@zinger/enums/lib/recurrence'
import { ServiceSegment } from '@zinger/enums/lib/service-segment'
import { range } from '@zinger/helpers'
import { createDwsServiceOptions } from '~/composables/create-service-options'
import { filterDwsProjectServiceMenuByCategory } from '~/composables/stores/use-dws-project-service-menu-resolver-store'
import { Contract } from '~/models/contract'
import { DwsProject, DwsProjectId } from '~/models/dws-project'
import { $datetime } from '~/services/datetime-service'
import { createContractStub } from '~~/stubs/create-contract-stub'
import { createDwsProjectServiceMenuStubs } from '~~/stubs/create-dws-project-service-menu-stub'
import { PROJECT_ID_MIN } from '~~/stubs/create-ltcs-project-stub'
import { createOwnExpenseProgramIndexResponseStub } from '~~/stubs/create-own-expense-program-index-response-stub'
import { STAFF_ID_MAX, STAFF_ID_MIN } from '~~/stubs/create-staff-stub'
import { createFaker } from '~~/stubs/fake'
import ramenIpsum from '~~/stubs/fake/ramen-ipsum'
import { STUB_DEFAULT_SEED } from '~~/stubs/index'

export function createDwsProjectStub (
  id: DwsProjectId = PROJECT_ID_MIN,
  contract: Contract = createContractStub(id)
): DwsProject {
  const seed = STUB_DEFAULT_SEED + `${id}`.padStart(8, '0')
  const faker = createFaker(seed)
  const ramen = ramenIpsum.factory(seed)
  const userId = contract.userId
  const menus = filterDwsProjectServiceMenuByCategory(createDwsProjectServiceMenuStubs())
  const ownExpense = createOwnExpenseProgramIndexResponseStub({ all: true })
  return {
    id,
    contractId: contract.id,
    officeId: contract.userId === 1 ? 8 : contract.officeId,
    userId,
    staffId: contract.userId === 1 ? 6 : faker.intBetween(STAFF_ID_MIN, STAFF_ID_MAX),
    writtenOn: faker.randomDateString(),
    effectivatedOn: faker.randomDateString(),
    requestFromUser: ramen.ipsum(200),
    requestFromFamily: ramen.ipsum(200),
    objective: ramen.ipsum(200),
    programs: range(1, faker.intBetween(1, 5)).map(summaryIndex => {
      const category = faker.randomElement(DwsProjectServiceCategory.values)
      const categoryMenus = menus(category).map(x => x.value)
      const contents = range(1, faker.intBetween(1, 10)).map(() => ({
        menuId: faker.randomElement(categoryMenus),
        content: ramen.ipsum(10),
        duration: faker.randomBoolean() ? faker.intBetween(10, 60) : undefined,
        memo: faker.randomBoolean() ? ramen.ipsum(10) : ''
      }))
      const contentsDurationSum = contents.reduce((content, val) => content + (val.duration ?? 0), 0)
      const programDate = $datetime.parse(faker.randomDateTimeString())
      const slot = {
        start: programDate.toFormat('HH:mm'),
        end: programDate.plus({ minutes: contentsDurationSum }).toFormat('HH:mm')
      }
      const options = createDwsServiceOptions('project', category).map(v => v.code)
      const ownExpenseProgramId = category === DwsProjectServiceCategory.ownExpense
        ? faker.randomElement(ownExpense.list.filter(x => x.officeId === contract.officeId))?.id
        : undefined
      return {
        summaryIndex,
        category,
        recurrence: faker.randomElement(Recurrence.values),
        dayOfWeeks: faker.randomArray(DayOfWeek.values, faker.intBetween(1, 7)),
        slot,
        headcount: faker.intBetween(1, 2) as 1 | 2,
        ownExpenseProgramId,
        options: faker.randomArray(options, faker.intBetween(1, options.length)),
        contents,
        note: faker.randomBoolean() ? ramen.ipsum(10) : ''
      }
    }),
    isEnabled: true,
    version: id,
    createdAt: faker.randomDateTimeString(),
    updatedAt: faker.randomDateTimeString()
  }
}

export function createDwsProjectStubs (xs: Contract[]): DwsProject[] {
  return xs
    .filter(contract => contract.serviceSegment === ServiceSegment.disabilitiesWelfare)
    .map(contract => createDwsProjectStub(contract.id, contract))
}
