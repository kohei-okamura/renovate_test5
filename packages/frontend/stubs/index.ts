/*
 * Copyright Â© 2019 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { createFaker } from '~~/stubs/fake'

export const ID_MAX = 100
export const ID_MIN = 1
export const STUB_COUNT = 1000
export const STUB_DEFAULT_SEED = 'zinger'

export const createSeeds = (n: number, seed: string = STUB_DEFAULT_SEED): string[] => {
  const faker = createFaker(seed)
  return Array.from(Array(n), () => faker.randomString(8))
}

export const SEEDS = createSeeds(STUB_COUNT * 3)

export type CreateStubs<T> = {
  (n?: number, skip?: number): T[]
}
