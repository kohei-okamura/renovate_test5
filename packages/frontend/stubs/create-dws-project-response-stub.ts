/*
 * Copyright Â© 2020 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { DwsProjectId } from '~/models/dws-project'
import { DwsProjectsApi } from '~/services/api/dws-projects-api'
import { createContractStub } from '~~/stubs/create-contract-stub'
import { createDwsProjectStub } from '~~/stubs/create-dws-project-stub'
import { USER_ID_MIN } from '~~/stubs/create-user-stub'

export const DWS_PROJECT_ID_MIN = USER_ID_MIN * 10

export function createDwsProjectResponseStub (id: DwsProjectId = DWS_PROJECT_ID_MIN): DwsProjectsApi.GetResponse {
  const contract = createContractStub(id)
  return {
    dwsProject: createDwsProjectStub(id, contract)
  }
}
