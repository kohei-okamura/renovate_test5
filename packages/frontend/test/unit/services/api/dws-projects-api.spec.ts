/*
 * Copyright © 2021 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import Axios from 'axios'
import { HttpStatusCode } from '~/models/http-status-code'
import { DwsProjectsApi } from '~/services/api/dws-projects-api'
import { createContractStub } from '~~/stubs/create-contract-stub'
import { createDwsProjectResponseStub } from '~~/stubs/create-dws-project-response-stub'
import { createDwsProjectStub } from '~~/stubs/create-dws-project-stub'
import { createAxiosMockAdapter } from '~~/test/helpers/create-axios-mock-adapter'
import { deleteUndefinedProperties } from '~~/test/helpers/delete-undefined-properties'

describe('api/dws-projects-api', () => {
  const axios = Axios.create()
  const adapter = createAxiosMockAdapter(axios)
  let dwsProjects: DwsProjectsApi.Definition

  beforeEach(() => {
    dwsProjects = DwsProjectsApi.create(axios)
  })

  const form: DwsProjectsApi.Form = {
    officeId: 16,
    staffId: 70,
    writtenOn: '1972-07-22',
    effectivatedOn: '2014-10-20',
    requestFromUser: '中太麺だった。もやしに中太麺が好きだ。太麺も好きだ。細麺は高菜がほうれん草がちぢれ麺だった。細麺とサンマーメンにもやしの野菜が好きだ。魚粉がつけめんマシマシで。あっさりが好きだ。',
    requestFromFamily: 'わかめの替え玉の細麺も好きだ。太麺は味玉とこってりが好きだ。こってりはニンニクが味噌の味玉がちぢれ麺のわかめとチャーシューメンと細麺が好きだ。ネギが好きだ。高菜に太麺も好きだ。',
    objective: 'もやしに中太麺が好きだ。太麺も好きだ。細麺は高菜がほうれん草がちぢれ麺だった。細麺とサンマーメンにもやしの野菜が好きだ。魚粉がつけめんマシマシで。あっさりが好きだ。',
    programs: [
      {
        summaryIndex: 1,
        dayOfWeeks: [1, 2, 6, 7],
        category: 11,
        recurrence: 24,
        slot: { start: '04:30', end: '12:00' },
        options: [300003, 301102, 301104, 301105, 301106],
        contents: [
          { menuId: 27, duration: 97, memo: '海苔も好きだ。' },
          { menuId: 89, duration: 51, memo: 'サンマーメンが好きだ。' }
        ]
      },
      {
        summaryIndex: 2,
        dayOfWeeks: [1, 2, 7],
        category: 11,
        recurrence: 24,
        slot: { start: '08:30', end: '16:30' },
        options: [300003, 301102, 301104, 301105, 301106],
        contents: [
          { menuId: 96, duration: 43, memo: '冷やし中華を注文する。' },
          { menuId: 46, duration: 92, memo: 'チャーシューを注文する。' },
          { menuId: 91, duration: 25, memo: '中太麺が好きだ。' },
          { menuId: 81, duration: 14, memo: 'メンマも好きだ。' }
        ]
      }
    ]
  }

  describe('create', () => {
    const userId = 0

    it('should post /api/users/:userId/dws-projects', async () => {
      const url = `/api/users/${userId}/dws-projects`
      adapter.setup(x => {
        x.onPost(url).replyOnce(HttpStatusCode.Created)
      })

      await dwsProjects.create({ form, userId })

      const request = adapter.getLastRequest()
      expect(request).toMatchObject({ method: 'post', url })
      expect(JSON.parse(request.data)).toStrictEqual(form)
    })

    it('should throw an AxiosError when failed to request', async () => {
      adapter.setup(x => {
        x.onPost(`/api/users/${userId}/dws-projects`).replyOnce(HttpStatusCode.BadRequest)
      })

      const promise = dwsProjects.create({ form, userId })

      await expect(promise).rejects.toThrowErrorMatchingSnapshot()
    })
  })

  describe('get', () => {
    it('should get /api/users/:userId/dws-projects/:id', async () => {
      const id = 1
      const userId = 10
      const url = `/api/users/${userId}/dws-projects/${id}`
      adapter.setup(x => {
        x.onGet(url).replyOnce(HttpStatusCode.OK, createDwsProjectResponseStub(id))
      })

      await dwsProjects.get({ id, userId })

      const request = adapter.getLastRequest()
      expect(request).toMatchObject({ method: 'get', url })
    })

    it('should return response of the api', async () => {
      const stub = createDwsProjectStub()
      const id = stub.id
      const userId = 10
      const expected = createDwsProjectResponseStub(id)
      adapter.setup(x => {
        x.onGet(`/api/users/${userId}/dws-projects/${id}`).replyOnce(HttpStatusCode.OK, expected)
      })

      const response = await dwsProjects.get({ id, userId })

      expect(response).toStrictEqual(deleteUndefinedProperties(expected))
    })

    it('should throw an AxiosError when failed to request', async () => {
      const id = 3
      const userId = 12
      adapter.setup(x => {
        x.onGet(`/api/users/${userId}/dws-projects/${id}`).replyOnce(HttpStatusCode.BadRequest)
      })

      const promise = dwsProjects.get({ id, userId })

      await expect(promise).rejects.toThrowErrorMatchingSnapshot()
    })
  })

  describe('update', () => {
    const userId = 1

    it('should put /api/users/:userId/dws-projects/:id', async () => {
      const id = 1
      const url = `/api/users/${userId}/dws-projects/${id}`
      adapter.setup(x => {
        x.onPut(url).replyOnce(HttpStatusCode.NoContent)
      })

      await dwsProjects.update({ form, id, userId })

      const request = adapter.getLastRequest()
      expect(request).toMatchObject({ method: 'put', url })
      expect(JSON.parse(request.data)).toStrictEqual(form)
    })

    it('should return response of the api', async () => {
      const id = 1
      const userId = 1
      const contractId = 100
      const current = createDwsProjectResponseStub()
      const dwsProject = createDwsProjectStub(current.dwsProject.id, createContractStub(contractId))
      const expected = { dwsProject }
      const form = {
        officeId: dwsProject.officeId,
        staffId: dwsProject.staffId,
        writtenOn: dwsProject.writtenOn,
        effectivatedOn: dwsProject.effectivatedOn,
        requestFromUser: dwsProject.requestFromUser,
        requestFromFamily: dwsProject.requestFromFamily,
        programs: [...dwsProject.programs],
        objective: dwsProject.objective
      }

      adapter.setup(x => {
        x.onPut(`/api/users/${userId}/dws-projects/${id}`).replyOnce(HttpStatusCode.OK, expected)
      })

      const response = await dwsProjects.update({ id, form, userId })

      expect(response).toStrictEqual(deleteUndefinedProperties(expected))
    })

    it('should throw an AxiosError when failed to request', async () => {
      const id = 2
      adapter.setup(x => {
        x.onPut(`/api/users/${userId}/dws-projects/${id}`).replyOnce(HttpStatusCode.BadRequest)
      })

      const promise = dwsProjects.update({ form, id, userId })

      await expect(promise).rejects.toThrowErrorMatchingSnapshot()
    })
  })
})
