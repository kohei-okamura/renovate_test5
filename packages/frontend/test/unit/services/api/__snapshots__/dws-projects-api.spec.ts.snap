// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api/dws-projects-api create should throw an AxiosError when failed to request 1`] = `"Request failed with status code 400"`;

exports[`api/dws-projects-api get should throw an AxiosError when failed to request 1`] = `"Request failed with status code 400"`;

exports[`api/dws-projects-api update should throw an AxiosError when failed to request 1`] = `"Request failed with status code 400"`;
